In [50]:
import pandas as pd
import numpy as np

In [51]:
data=pd.read_csv("W3(data).csv")
data.head()

Suburb             Address  Rooms Type      Price Method SellerG  \
0  Abbotsford       68 Studley St    2.0    h        NaN     SS  Jellis   
1  Abbotsford        85 Turner St    2.0    h  1480000.0      S  Biggin   
2  Abbotsford     25 Bloomburg St    2.0    h  1035000.0      S  Biggin   
3  Abbotsford  18/659 Victoria St    3.0    u        NaN     VB  Rounds   
4  Abbotsford        5 Charles St    3.0    h  1465000.0     SP  Biggin   

         Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  03-09-2016       2.5    3067.0  ...       1.0  1.0     126.0           NaN   
1  03-12-2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
2  04-02-2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
3  04-02-2016       2.5    3067.0  ...       2.0  1.0       0.0           NaN   
4  04-03-2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.8014    144.9958  Northern Metropolitan   
1        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
2     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
3        NaN        Yarra  -37.8114    145.0116  Northern Metropolitan   
4     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

In [52]:
prev=data.copy()

#### 1.Determine the no. of rows and no. of columns in the dataset.

In [53]:
# computing number of rows 
rows = len(data.axes[0]) 
print("Number of Rows: ", rows) 

# computing number of columns 
cols = len(data.axes[1])
print("Number of Columns: ", cols)

Number of Rows:  23678
Number of Columns:  21


##### 2. Display the datatypes of each column.

In [54]:
# return the dtype of each column 
result = data.dtypes   
# Print the result 
print(result) 

Suburb            object
Address           object
Rooms            float64
Type              object
Price            float64
Method            object
SellerG           object
Date              object
Distance         float64
Postcode         float64
Bedroom2         float64
Bathroom         float64
Car              float64
Landsize         float64
BuildingArea     float64
YearBuilt        float64
CouncilArea       object
Lattitude        float64
Longtitude       float64
Regionname        object
Propertycount    float64
dtype: object


#### 3. Determine the row indexes of rows having all null values (i.e. empty rows)

In [55]:
rows_with_NAN=[index for index, row in data.iterrows() if row.isnull().any()]
print(rows_with_NAN)

[0, 1, 3, 5, 7, 8, 9, 10, 12, 13, 15, 16, 17, 19, 20, 21, 22, 23, 26, 27, 28, 29, 31, 32, 34, 35, 37, 40, 41, 42, 46, 47, 48, 49, 51, 53, 54, 55, 56, 61, 63, 65, 66, 70, 75, 76, 77, 78, 79, 80, 81, 85, 88, 90, 92, 93, 94, 95, 96, 97, 98, 100, 102, 103, 106, 108, 111, 112, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 131, 132, 133, 135, 138, 140, 141, 142, 144, 145, 147, 149, 151, 152, 154, 155, 157, 158, 159, 161, 164, 165, 166, 168, 170, 171, 172, 174, 175, 176, 180, 181, 182, 183, 185, 186, 188, 189, 190, 191, 193, 195, 197, 198, 199, 200, 201, 204, 207, 208, 209, 210, 211, 212, 213, 214, 219, 221, 222, 223, 224, 225, 226, 228, 229, 235, 238, 240, 241, 244, 246, 248, 249, 251, 253, 254, 255, 257, 258, 259, 261, 264, 266, 267, 268, 269, 271, 272, 273, 274, 275, 276, 278, 280, 281, 282, 284, 287, 288, 289, 291, 292, 294, 295, 297, 298, 299, 301, 303, 304, 305, 306, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 324, 325, 326, 327, 328, 32

#### 4. Determine the total number of empty rows and their percentage in whole dataset and remove them.

In [39]:
#total percentage of empty rows in whole dataset 
per=((data.isnull().sum() / data.shape[0]) * 100.00).round(3)
print(per)

Suburb            0.553
Address           0.553
Rooms             0.553
Type              0.553
Price            22.308
Method            0.553
SellerG           0.553
Date              0.553
Distance          0.557
Postcode          0.557
Bedroom2         19.478
Bathroom         19.491
Car              20.090
Landsize         26.472
BuildingArea     57.691
YearBuilt        51.263
CouncilArea      33.880
Lattitude        18.730
Longtitude       18.730
Regionname        0.557
Propertycount     0.557
dtype: float64


In [3]:
# dropping column with all null values 
new=data.dropna(axis = 1, how ='all', inplace = True) 


#### 5. Determine the column names having missing values.

In [59]:
nl=data.isnull().sum()
print("columns name missing values:\n",df5.index.tolist())

columns name missing values:
 ['Suburb', 'Address', 'Rooms', 'Type', 'Price', 'Method', 'SellerG', 'Date', 'Distance', 'Postcode', 'Bedroom2', 'Bathroom', 'Car', 'Landsize', 'BuildingArea', 'YearBuilt', 'CouncilArea', 'Lattitude', 'Longtitude', 'Regionname', 'Propertycount']


#### 6. Determine the missing values and their percentage in each column.

In [60]:
nl=data.isnull().sum().index.tolist()
for i in df6:
    null_values=data[i].isnull().sum()
    percentage=round(((null_values/data.shape[0])*100),2)
    print(i,"              ",null_values,"             ",     percentage,'%')

Suburb                131               0.55 %
Address                131               0.55 %
Rooms                131               0.55 %
Type                131               0.55 %
Price                5282               22.31 %
Method                131               0.55 %
SellerG                131               0.55 %
Date                131               0.55 %
Distance                132               0.56 %
Postcode                132               0.56 %
Bedroom2                4612               19.48 %
Bathroom                4615               19.49 %
Car                4757               20.09 %
Landsize                6268               26.47 %
BuildingArea                13660               57.69 %
YearBuilt                12138               51.26 %
CouncilArea                8022               33.88 %
Lattitude                4435               18.73 %
Longtitude                4435               18.73 %
Regionname                132               0.56 %
Propertyco

#### 7. Determine the row indexes having missing values.

In [61]:
data.index[data.isna().any(axis=1)]

Int64Index([    0,     1,     3,     5,     7,     8,     9,    10,    12,
               13,
            ...
            23668, 23669, 23670, 23671, 23672, 23673, 23674, 23675, 23676,
            23677],
           dtype='int64', length=17482)

#### 8. Determine the missing values and their percentage in each row. 

In [62]:
count=data.apply(lambda x: abs(x.count()-data.shape[1]), axis=1)    
row_id=count.index.tolist()
values=count.values.tolist() #null values
for i in range(len(row_id)):
    if values[i]>0:
        null_values=values[i]
        percentage=round(((null_values/data.shape[1])*100),2)
        print('Row ',row_id[i],' no. of null values =',null_values,'percentage = ',percentage)

Row  0  no. of null values = 3 percentage =  14.29
Row  1  no. of null values = 2 percentage =  9.52
Row  3  no. of null values = 3 percentage =  14.29
Row  5  no. of null values = 2 percentage =  9.52
Row  7  no. of null values = 1 percentage =  4.76
Row  8  no. of null values = 2 percentage =  9.52
Row  9  no. of null values = 2 percentage =  9.52
Row  10  no. of null values = 2 percentage =  9.52
Row  12  no. of null values = 1 percentage =  4.76
Row  13  no. of null values = 1 percentage =  4.76
Row  15  no. of null values = 9 percentage =  42.86
Row  16  no. of null values = 9 percentage =  42.86
Row  17  no. of null values = 2 percentage =  9.52
Row  19  no. of null values = 9 percentage =  42.86
Row  20  no. of null values = 1 percentage =  4.76
Row  21  no. of null values = 9 percentage =  42.86
Row  22  no. of null values = 9 percentage =  42.86
Row  23  no. of null values = 2 percentage =  9.52
Row  26  no. of null values = 7 percentage =  33.33
Row  27  no. of null values = 

Row  426  no. of null values = 2 percentage =  9.52
Row  428  no. of null values = 1 percentage =  4.76
Row  429  no. of null values = 2 percentage =  9.52
Row  430  no. of null values = 2 percentage =  9.52
Row  431  no. of null values = 2 percentage =  9.52
Row  432  no. of null values = 2 percentage =  9.52
Row  434  no. of null values = 10 percentage =  47.62
Row  435  no. of null values = 2 percentage =  9.52
Row  436  no. of null values = 1 percentage =  4.76
Row  437  no. of null values = 1 percentage =  4.76
Row  438  no. of null values = 2 percentage =  9.52
Row  440  no. of null values = 1 percentage =  4.76
Row  441  no. of null values = 2 percentage =  9.52
Row  443  no. of null values = 10 percentage =  47.62
Row  444  no. of null values = 10 percentage =  47.62
Row  447  no. of null values = 3 percentage =  14.29
Row  448  no. of null values = 3 percentage =  14.29
Row  449  no. of null values = 3 percentage =  14.29
Row  450  no. of null values = 2 percentage =  9.52
Row

Row  997  no. of null values = 10 percentage =  47.62
Row  998  no. of null values = 2 percentage =  9.52
Row  1000  no. of null values = 2 percentage =  9.52
Row  1001  no. of null values = 10 percentage =  47.62
Row  1003  no. of null values = 9 percentage =  42.86
Row  1004  no. of null values = 1 percentage =  4.76
Row  1005  no. of null values = 2 percentage =  9.52
Row  1007  no. of null values = 1 percentage =  4.76
Row  1008  no. of null values = 1 percentage =  4.76
Row  1010  no. of null values = 2 percentage =  9.52
Row  1014  no. of null values = 2 percentage =  9.52
Row  1015  no. of null values = 9 percentage =  42.86
Row  1016  no. of null values = 9 percentage =  42.86
Row  1017  no. of null values = 2 percentage =  9.52
Row  1022  no. of null values = 1 percentage =  4.76
Row  1023  no. of null values = 1 percentage =  4.76
Row  1024  no. of null values = 9 percentage =  42.86
Row  1027  no. of null values = 1 percentage =  4.76
Row  1028  no. of null values = 10 perce

Row  1354  no. of null values = 1 percentage =  4.76
Row  1356  no. of null values = 2 percentage =  9.52
Row  1357  no. of null values = 1 percentage =  4.76
Row  1358  no. of null values = 1 percentage =  4.76
Row  1359  no. of null values = 10 percentage =  47.62
Row  1360  no. of null values = 2 percentage =  9.52
Row  1361  no. of null values = 9 percentage =  42.86
Row  1362  no. of null values = 9 percentage =  42.86
Row  1363  no. of null values = 2 percentage =  9.52
Row  1366  no. of null values = 9 percentage =  42.86
Row  1367  no. of null values = 2 percentage =  9.52
Row  1368  no. of null values = 9 percentage =  42.86
Row  1370  no. of null values = 9 percentage =  42.86
Row  1371  no. of null values = 2 percentage =  9.52
Row  1372  no. of null values = 9 percentage =  42.86
Row  1373  no. of null values = 2 percentage =  9.52
Row  1374  no. of null values = 9 percentage =  42.86
Row  1375  no. of null values = 1 percentage =  4.76
Row  1376  no. of null values = 1 per

Row  1720  no. of null values = 1 percentage =  4.76
Row  1722  no. of null values = 1 percentage =  4.76
Row  1723  no. of null values = 9 percentage =  42.86
Row  1724  no. of null values = 2 percentage =  9.52
Row  1725  no. of null values = 1 percentage =  4.76
Row  1726  no. of null values = 9 percentage =  42.86
Row  1730  no. of null values = 2 percentage =  9.52
Row  1732  no. of null values = 2 percentage =  9.52
Row  1734  no. of null values = 2 percentage =  9.52
Row  1735  no. of null values = 1 percentage =  4.76
Row  1736  no. of null values = 1 percentage =  4.76
Row  1738  no. of null values = 3 percentage =  14.29
Row  1739  no. of null values = 1 percentage =  4.76
Row  1740  no. of null values = 2 percentage =  9.52
Row  1741  no. of null values = 9 percentage =  42.86
Row  1743  no. of null values = 1 percentage =  4.76
Row  1744  no. of null values = 1 percentage =  4.76
Row  1746  no. of null values = 1 percentage =  4.76
Row  1748  no. of null values = 2 percenta

Row  2089  no. of null values = 1 percentage =  4.76
Row  2090  no. of null values = 2 percentage =  9.52
Row  2091  no. of null values = 9 percentage =  42.86
Row  2092  no. of null values = 2 percentage =  9.52
Row  2093  no. of null values = 2 percentage =  9.52
Row  2094  no. of null values = 3 percentage =  14.29
Row  2095  no. of null values = 1 percentage =  4.76
Row  2096  no. of null values = 1 percentage =  4.76
Row  2098  no. of null values = 2 percentage =  9.52
Row  2100  no. of null values = 1 percentage =  4.76
Row  2101  no. of null values = 3 percentage =  14.29
Row  2102  no. of null values = 7 percentage =  33.33
Row  2103  no. of null values = 10 percentage =  47.62
Row  2105  no. of null values = 1 percentage =  4.76
Row  2106  no. of null values = 1 percentage =  4.76
Row  2107  no. of null values = 2 percentage =  9.52
Row  2108  no. of null values = 2 percentage =  9.52
Row  2109  no. of null values = 10 percentage =  47.62
Row  2110  no. of null values = 9 perc

Row  2424  no. of null values = 9 percentage =  42.86
Row  2426  no. of null values = 1 percentage =  4.76
Row  2427  no. of null values = 9 percentage =  42.86
Row  2428  no. of null values = 1 percentage =  4.76
Row  2430  no. of null values = 2 percentage =  9.52
Row  2433  no. of null values = 1 percentage =  4.76
Row  2435  no. of null values = 9 percentage =  42.86
Row  2438  no. of null values = 2 percentage =  9.52
Row  2440  no. of null values = 9 percentage =  42.86
Row  2441  no. of null values = 1 percentage =  4.76
Row  2442  no. of null values = 9 percentage =  42.86
Row  2444  no. of null values = 2 percentage =  9.52
Row  2446  no. of null values = 2 percentage =  9.52
Row  2448  no. of null values = 2 percentage =  9.52
Row  2450  no. of null values = 2 percentage =  9.52
Row  2451  no. of null values = 3 percentage =  14.29
Row  2452  no. of null values = 1 percentage =  4.76
Row  2453  no. of null values = 1 percentage =  4.76
Row  2456  no. of null values = 9 percen

Row  2696  no. of null values = 9 percentage =  42.86
Row  2697  no. of null values = 10 percentage =  47.62
Row  2698  no. of null values = 1 percentage =  4.76
Row  2699  no. of null values = 1 percentage =  4.76
Row  2700  no. of null values = 1 percentage =  4.76
Row  2701  no. of null values = 1 percentage =  4.76
Row  2702  no. of null values = 3 percentage =  14.29
Row  2704  no. of null values = 9 percentage =  42.86
Row  2705  no. of null values = 1 percentage =  4.76
Row  2706  no. of null values = 1 percentage =  4.76
Row  2707  no. of null values = 9 percentage =  42.86
Row  2709  no. of null values = 10 percentage =  47.62
Row  2710  no. of null values = 3 percentage =  14.29
Row  2711  no. of null values = 9 percentage =  42.86
Row  2713  no. of null values = 1 percentage =  4.76
Row  2714  no. of null values = 1 percentage =  4.76
Row  2716  no. of null values = 10 percentage =  47.62
Row  2719  no. of null values = 10 percentage =  47.62
Row  2720  no. of null values = 

Row  2996  no. of null values = 2 percentage =  9.52
Row  2998  no. of null values = 9 percentage =  42.86
Row  2999  no. of null values = 9 percentage =  42.86
Row  3000  no. of null values = 7 percentage =  33.33
Row  3001  no. of null values = 2 percentage =  9.52
Row  3002  no. of null values = 3 percentage =  14.29
Row  3003  no. of null values = 3 percentage =  14.29
Row  3004  no. of null values = 9 percentage =  42.86
Row  3007  no. of null values = 9 percentage =  42.86
Row  3011  no. of null values = 2 percentage =  9.52
Row  3012  no. of null values = 1 percentage =  4.76
Row  3013  no. of null values = 1 percentage =  4.76
Row  3014  no. of null values = 2 percentage =  9.52
Row  3015  no. of null values = 1 percentage =  4.76
Row  3016  no. of null values = 1 percentage =  4.76
Row  3019  no. of null values = 10 percentage =  47.62
Row  3020  no. of null values = 2 percentage =  9.52
Row  3022  no. of null values = 1 percentage =  4.76
Row  3023  no. of null values = 2 per

Row  3405  no. of null values = 2 percentage =  9.52
Row  3410  no. of null values = 2 percentage =  9.52
Row  3413  no. of null values = 1 percentage =  4.76
Row  3415  no. of null values = 2 percentage =  9.52
Row  3417  no. of null values = 1 percentage =  4.76
Row  3418  no. of null values = 1 percentage =  4.76
Row  3420  no. of null values = 2 percentage =  9.52
Row  3421  no. of null values = 9 percentage =  42.86
Row  3422  no. of null values = 3 percentage =  14.29
Row  3424  no. of null values = 9 percentage =  42.86
Row  3426  no. of null values = 1 percentage =  4.76
Row  3427  no. of null values = 1 percentage =  4.76
Row  3428  no. of null values = 1 percentage =  4.76
Row  3431  no. of null values = 2 percentage =  9.52
Row  3433  no. of null values = 3 percentage =  14.29
Row  3436  no. of null values = 1 percentage =  4.76
Row  3437  no. of null values = 1 percentage =  4.76
Row  3438  no. of null values = 2 percentage =  9.52
Row  3439  no. of null values = 2 percenta

Row  3730  no. of null values = 1 percentage =  4.76
Row  3735  no. of null values = 2 percentage =  9.52
Row  3737  no. of null values = 1 percentage =  4.76
Row  3739  no. of null values = 2 percentage =  9.52
Row  3742  no. of null values = 1 percentage =  4.76
Row  3744  no. of null values = 9 percentage =  42.86
Row  3745  no. of null values = 21 percentage =  100.0
Row  3748  no. of null values = 9 percentage =  42.86
Row  3749  no. of null values = 3 percentage =  14.29
Row  3751  no. of null values = 9 percentage =  42.86
Row  3752  no. of null values = 21 percentage =  100.0
Row  3755  no. of null values = 2 percentage =  9.52
Row  3757  no. of null values = 1 percentage =  4.76
Row  3758  no. of null values = 1 percentage =  4.76
Row  3759  no. of null values = 2 percentage =  9.52
Row  3760  no. of null values = 1 percentage =  4.76
Row  3761  no. of null values = 7 percentage =  33.33
Row  3765  no. of null values = 1 percentage =  4.76
Row  3766  no. of null values = 1 per

Row  4033  no. of null values = 9 percentage =  42.86
Row  4035  no. of null values = 2 percentage =  9.52
Row  4036  no. of null values = 2 percentage =  9.52
Row  4038  no. of null values = 1 percentage =  4.76
Row  4039  no. of null values = 2 percentage =  9.52
Row  4040  no. of null values = 2 percentage =  9.52
Row  4042  no. of null values = 2 percentage =  9.52
Row  4045  no. of null values = 2 percentage =  9.52
Row  4046  no. of null values = 2 percentage =  9.52
Row  4047  no. of null values = 1 percentage =  4.76
Row  4048  no. of null values = 2 percentage =  9.52
Row  4049  no. of null values = 9 percentage =  42.86
Row  4050  no. of null values = 2 percentage =  9.52
Row  4054  no. of null values = 2 percentage =  9.52
Row  4055  no. of null values = 2 percentage =  9.52
Row  4058  no. of null values = 2 percentage =  9.52
Row  4059  no. of null values = 2 percentage =  9.52
Row  4061  no. of null values = 2 percentage =  9.52
Row  4063  no. of null values = 1 percentage

Row  4417  no. of null values = 2 percentage =  9.52
Row  4418  no. of null values = 2 percentage =  9.52
Row  4420  no. of null values = 2 percentage =  9.52
Row  4423  no. of null values = 1 percentage =  4.76
Row  4424  no. of null values = 1 percentage =  4.76
Row  4425  no. of null values = 1 percentage =  4.76
Row  4429  no. of null values = 2 percentage =  9.52
Row  4431  no. of null values = 2 percentage =  9.52
Row  4432  no. of null values = 9 percentage =  42.86
Row  4435  no. of null values = 2 percentage =  9.52
Row  4436  no. of null values = 1 percentage =  4.76
Row  4437  no. of null values = 2 percentage =  9.52
Row  4445  no. of null values = 1 percentage =  4.76
Row  4447  no. of null values = 1 percentage =  4.76
Row  4448  no. of null values = 2 percentage =  9.52
Row  4450  no. of null values = 3 percentage =  14.29
Row  4452  no. of null values = 1 percentage =  4.76
Row  4453  no. of null values = 2 percentage =  9.52
Row  4454  no. of null values = 1 percentage

Row  4697  no. of null values = 2 percentage =  9.52
Row  4698  no. of null values = 1 percentage =  4.76
Row  4699  no. of null values = 1 percentage =  4.76
Row  4700  no. of null values = 2 percentage =  9.52
Row  4701  no. of null values = 2 percentage =  9.52
Row  4702  no. of null values = 1 percentage =  4.76
Row  4703  no. of null values = 2 percentage =  9.52
Row  4704  no. of null values = 3 percentage =  14.29
Row  4705  no. of null values = 10 percentage =  47.62
Row  4710  no. of null values = 2 percentage =  9.52
Row  4711  no. of null values = 10 percentage =  47.62
Row  4714  no. of null values = 10 percentage =  47.62
Row  4716  no. of null values = 2 percentage =  9.52
Row  4717  no. of null values = 2 percentage =  9.52
Row  4721  no. of null values = 10 percentage =  47.62
Row  4723  no. of null values = 10 percentage =  47.62
Row  4724  no. of null values = 10 percentage =  47.62
Row  4725  no. of null values = 9 percentage =  42.86
Row  4726  no. of null values = 

Row  5073  no. of null values = 1 percentage =  4.76
Row  5074  no. of null values = 2 percentage =  9.52
Row  5075  no. of null values = 9 percentage =  42.86
Row  5077  no. of null values = 2 percentage =  9.52
Row  5079  no. of null values = 1 percentage =  4.76
Row  5080  no. of null values = 10 percentage =  47.62
Row  5082  no. of null values = 2 percentage =  9.52
Row  5083  no. of null values = 9 percentage =  42.86
Row  5084  no. of null values = 1 percentage =  4.76
Row  5085  no. of null values = 1 percentage =  4.76
Row  5086  no. of null values = 1 percentage =  4.76
Row  5088  no. of null values = 9 percentage =  42.86
Row  5089  no. of null values = 1 percentage =  4.76
Row  5091  no. of null values = 9 percentage =  42.86
Row  5092  no. of null values = 1 percentage =  4.76
Row  5093  no. of null values = 2 percentage =  9.52
Row  5095  no. of null values = 2 percentage =  9.52
Row  5096  no. of null values = 9 percentage =  42.86
Row  5098  no. of null values = 9 perce

Row  5319  no. of null values = 2 percentage =  9.52
Row  5320  no. of null values = 3 percentage =  14.29
Row  5321  no. of null values = 1 percentage =  4.76
Row  5323  no. of null values = 1 percentage =  4.76
Row  5324  no. of null values = 10 percentage =  47.62
Row  5326  no. of null values = 1 percentage =  4.76
Row  5327  no. of null values = 3 percentage =  14.29
Row  5328  no. of null values = 21 percentage =  100.0
Row  5329  no. of null values = 1 percentage =  4.76
Row  5330  no. of null values = 10 percentage =  47.62
Row  5331  no. of null values = 21 percentage =  100.0
Row  5332  no. of null values = 2 percentage =  9.52
Row  5333  no. of null values = 10 percentage =  47.62
Row  5335  no. of null values = 1 percentage =  4.76
Row  5338  no. of null values = 1 percentage =  4.76
Row  5342  no. of null values = 2 percentage =  9.52
Row  5343  no. of null values = 1 percentage =  4.76
Row  5345  no. of null values = 1 percentage =  4.76
Row  5346  no. of null values = 1 

Row  5681  no. of null values = 1 percentage =  4.76
Row  5684  no. of null values = 2 percentage =  9.52
Row  5688  no. of null values = 2 percentage =  9.52
Row  5689  no. of null values = 2 percentage =  9.52
Row  5691  no. of null values = 9 percentage =  42.86
Row  5694  no. of null values = 1 percentage =  4.76
Row  5695  no. of null values = 1 percentage =  4.76
Row  5697  no. of null values = 3 percentage =  14.29
Row  5698  no. of null values = 2 percentage =  9.52
Row  5700  no. of null values = 9 percentage =  42.86
Row  5701  no. of null values = 2 percentage =  9.52
Row  5703  no. of null values = 9 percentage =  42.86
Row  5705  no. of null values = 1 percentage =  4.76
Row  5706  no. of null values = 9 percentage =  42.86
Row  5708  no. of null values = 10 percentage =  47.62
Row  5711  no. of null values = 9 percentage =  42.86
Row  5713  no. of null values = 2 percentage =  9.52
Row  5715  no. of null values = 1 percentage =  4.76
Row  5719  no. of null values = 2 perc

Row  5990  no. of null values = 9 percentage =  42.86
Row  5991  no. of null values = 1 percentage =  4.76
Row  5992  no. of null values = 1 percentage =  4.76
Row  5993  no. of null values = 9 percentage =  42.86
Row  5997  no. of null values = 1 percentage =  4.76
Row  5998  no. of null values = 2 percentage =  9.52
Row  6001  no. of null values = 2 percentage =  9.52
Row  6002  no. of null values = 2 percentage =  9.52
Row  6003  no. of null values = 1 percentage =  4.76
Row  6004  no. of null values = 1 percentage =  4.76
Row  6005  no. of null values = 3 percentage =  14.29
Row  6006  no. of null values = 10 percentage =  47.62
Row  6010  no. of null values = 1 percentage =  4.76
Row  6011  no. of null values = 1 percentage =  4.76
Row  6012  no. of null values = 3 percentage =  14.29
Row  6013  no. of null values = 9 percentage =  42.86
Row  6014  no. of null values = 7 percentage =  33.33
Row  6016  no. of null values = 10 percentage =  47.62
Row  6017  no. of null values = 3 pe

Row  6327  no. of null values = 3 percentage =  14.29
Row  6328  no. of null values = 10 percentage =  47.62
Row  6329  no. of null values = 9 percentage =  42.86
Row  6332  no. of null values = 3 percentage =  14.29
Row  6333  no. of null values = 3 percentage =  14.29
Row  6334  no. of null values = 10 percentage =  47.62
Row  6335  no. of null values = 2 percentage =  9.52
Row  6336  no. of null values = 2 percentage =  9.52
Row  6337  no. of null values = 3 percentage =  14.29
Row  6339  no. of null values = 2 percentage =  9.52
Row  6342  no. of null values = 1 percentage =  4.76
Row  6343  no. of null values = 1 percentage =  4.76
Row  6344  no. of null values = 3 percentage =  14.29
Row  6345  no. of null values = 10 percentage =  47.62
Row  6346  no. of null values = 9 percentage =  42.86
Row  6348  no. of null values = 10 percentage =  47.62
Row  6350  no. of null values = 2 percentage =  9.52
Row  6352  no. of null values = 2 percentage =  9.52
Row  6353  no. of null values =

Row  6594  no. of null values = 9 percentage =  42.86
Row  6595  no. of null values = 9 percentage =  42.86
Row  6596  no. of null values = 2 percentage =  9.52
Row  6597  no. of null values = 9 percentage =  42.86
Row  6598  no. of null values = 9 percentage =  42.86
Row  6599  no. of null values = 9 percentage =  42.86
Row  6600  no. of null values = 9 percentage =  42.86
Row  6601  no. of null values = 9 percentage =  42.86
Row  6602  no. of null values = 10 percentage =  47.62
Row  6605  no. of null values = 1 percentage =  4.76
Row  6606  no. of null values = 9 percentage =  42.86
Row  6607  no. of null values = 1 percentage =  4.76
Row  6608  no. of null values = 2 percentage =  9.52
Row  6610  no. of null values = 2 percentage =  9.52
Row  6611  no. of null values = 2 percentage =  9.52
Row  6612  no. of null values = 2 percentage =  9.52
Row  6614  no. of null values = 9 percentage =  42.86
Row  6615  no. of null values = 10 percentage =  47.62
Row  6616  no. of null values = 9

Row  6991  no. of null values = 2 percentage =  9.52
Row  6994  no. of null values = 1 percentage =  4.76
Row  6996  no. of null values = 2 percentage =  9.52
Row  6999  no. of null values = 1 percentage =  4.76
Row  7001  no. of null values = 1 percentage =  4.76
Row  7004  no. of null values = 1 percentage =  4.76
Row  7005  no. of null values = 9 percentage =  42.86
Row  7006  no. of null values = 2 percentage =  9.52
Row  7008  no. of null values = 3 percentage =  14.29
Row  7009  no. of null values = 9 percentage =  42.86
Row  7011  no. of null values = 1 percentage =  4.76
Row  7013  no. of null values = 9 percentage =  42.86
Row  7014  no. of null values = 2 percentage =  9.52
Row  7016  no. of null values = 2 percentage =  9.52
Row  7021  no. of null values = 1 percentage =  4.76
Row  7022  no. of null values = 2 percentage =  9.52
Row  7024  no. of null values = 2 percentage =  9.52
Row  7025  no. of null values = 2 percentage =  9.52
Row  7026  no. of null values = 1 percenta

Row  7402  no. of null values = 1 percentage =  4.76
Row  7403  no. of null values = 3 percentage =  14.29
Row  7404  no. of null values = 1 percentage =  4.76
Row  7408  no. of null values = 2 percentage =  9.52
Row  7410  no. of null values = 2 percentage =  9.52
Row  7411  no. of null values = 2 percentage =  9.52
Row  7415  no. of null values = 1 percentage =  4.76
Row  7417  no. of null values = 2 percentage =  9.52
Row  7418  no. of null values = 1 percentage =  4.76
Row  7419  no. of null values = 9 percentage =  42.86
Row  7420  no. of null values = 9 percentage =  42.86
Row  7422  no. of null values = 9 percentage =  42.86
Row  7423  no. of null values = 1 percentage =  4.76
Row  7424  no. of null values = 1 percentage =  4.76
Row  7426  no. of null values = 2 percentage =  9.52
Row  7428  no. of null values = 2 percentage =  9.52
Row  7429  no. of null values = 1 percentage =  4.76
Row  7431  no. of null values = 2 percentage =  9.52
Row  7434  no. of null values = 1 percenta

Row  7636  no. of null values = 9 percentage =  42.86
Row  7637  no. of null values = 1 percentage =  4.76
Row  7638  no. of null values = 2 percentage =  9.52
Row  7639  no. of null values = 10 percentage =  47.62
Row  7640  no. of null values = 2 percentage =  9.52
Row  7642  no. of null values = 2 percentage =  9.52
Row  7643  no. of null values = 7 percentage =  33.33
Row  7644  no. of null values = 1 percentage =  4.76
Row  7645  no. of null values = 2 percentage =  9.52
Row  7647  no. of null values = 2 percentage =  9.52
Row  7648  no. of null values = 2 percentage =  9.52
Row  7649  no. of null values = 9 percentage =  42.86
Row  7650  no. of null values = 1 percentage =  4.76
Row  7651  no. of null values = 1 percentage =  4.76
Row  7655  no. of null values = 2 percentage =  9.52
Row  7657  no. of null values = 2 percentage =  9.52
Row  7658  no. of null values = 2 percentage =  9.52
Row  7659  no. of null values = 1 percentage =  4.76
Row  7660  no. of null values = 2 percent

Row  8019  no. of null values = 2 percentage =  9.52
Row  8020  no. of null values = 3 percentage =  14.29
Row  8021  no. of null values = 1 percentage =  4.76
Row  8024  no. of null values = 10 percentage =  47.62
Row  8025  no. of null values = 1 percentage =  4.76
Row  8027  no. of null values = 1 percentage =  4.76
Row  8028  no. of null values = 3 percentage =  14.29
Row  8029  no. of null values = 1 percentage =  4.76
Row  8030  no. of null values = 2 percentage =  9.52
Row  8031  no. of null values = 1 percentage =  4.76
Row  8032  no. of null values = 2 percentage =  9.52
Row  8033  no. of null values = 2 percentage =  9.52
Row  8036  no. of null values = 2 percentage =  9.52
Row  8037  no. of null values = 10 percentage =  47.62
Row  8039  no. of null values = 1 percentage =  4.76
Row  8041  no. of null values = 9 percentage =  42.86
Row  8042  no. of null values = 1 percentage =  4.76
Row  8043  no. of null values = 3 percentage =  14.29
Row  8044  no. of null values = 9 perc

Row  8367  no. of null values = 9 percentage =  42.86
Row  8369  no. of null values = 2 percentage =  9.52
Row  8372  no. of null values = 9 percentage =  42.86
Row  8373  no. of null values = 9 percentage =  42.86
Row  8374  no. of null values = 2 percentage =  9.52
Row  8376  no. of null values = 10 percentage =  47.62
Row  8377  no. of null values = 9 percentage =  42.86
Row  8378  no. of null values = 9 percentage =  42.86
Row  8379  no. of null values = 9 percentage =  42.86
Row  8380  no. of null values = 1 percentage =  4.76
Row  8382  no. of null values = 9 percentage =  42.86
Row  8384  no. of null values = 9 percentage =  42.86
Row  8385  no. of null values = 9 percentage =  42.86
Row  8390  no. of null values = 9 percentage =  42.86
Row  8391  no. of null values = 3 percentage =  14.29
Row  8392  no. of null values = 1 percentage =  4.76
Row  8393  no. of null values = 2 percentage =  9.52
Row  8396  no. of null values = 9 percentage =  42.86
Row  8398  no. of null values = 

Row  8636  no. of null values = 9 percentage =  42.86
Row  8637  no. of null values = 9 percentage =  42.86
Row  8638  no. of null values = 10 percentage =  47.62
Row  8641  no. of null values = 9 percentage =  42.86
Row  8644  no. of null values = 2 percentage =  9.52
Row  8645  no. of null values = 2 percentage =  9.52
Row  8648  no. of null values = 9 percentage =  42.86
Row  8649  no. of null values = 9 percentage =  42.86
Row  8650  no. of null values = 1 percentage =  4.76
Row  8651  no. of null values = 2 percentage =  9.52
Row  8652  no. of null values = 21 percentage =  100.0
Row  8653  no. of null values = 1 percentage =  4.76
Row  8654  no. of null values = 2 percentage =  9.52
Row  8655  no. of null values = 2 percentage =  9.52
Row  8656  no. of null values = 9 percentage =  42.86
Row  8657  no. of null values = 2 percentage =  9.52
Row  8658  no. of null values = 2 percentage =  9.52
Row  8659  no. of null values = 9 percentage =  42.86
Row  8660  no. of null values = 1 p

Row  8984  no. of null values = 1 percentage =  4.76
Row  8985  no. of null values = 2 percentage =  9.52
Row  8986  no. of null values = 2 percentage =  9.52
Row  8987  no. of null values = 10 percentage =  47.62
Row  8988  no. of null values = 3 percentage =  14.29
Row  8989  no. of null values = 2 percentage =  9.52
Row  8991  no. of null values = 9 percentage =  42.86
Row  8996  no. of null values = 1 percentage =  4.76
Row  8997  no. of null values = 2 percentage =  9.52
Row  8999  no. of null values = 7 percentage =  33.33
Row  9000  no. of null values = 1 percentage =  4.76
Row  9002  no. of null values = 9 percentage =  42.86
Row  9003  no. of null values = 2 percentage =  9.52
Row  9004  no. of null values = 2 percentage =  9.52
Row  9009  no. of null values = 1 percentage =  4.76
Row  9010  no. of null values = 2 percentage =  9.52
Row  9011  no. of null values = 2 percentage =  9.52
Row  9012  no. of null values = 2 percentage =  9.52
Row  9014  no. of null values = 2 percen

Row  9284  no. of null values = 1 percentage =  4.76
Row  9285  no. of null values = 1 percentage =  4.76
Row  9286  no. of null values = 9 percentage =  42.86
Row  9289  no. of null values = 1 percentage =  4.76
Row  9291  no. of null values = 1 percentage =  4.76
Row  9292  no. of null values = 1 percentage =  4.76
Row  9295  no. of null values = 2 percentage =  9.52
Row  9297  no. of null values = 1 percentage =  4.76
Row  9298  no. of null values = 2 percentage =  9.52
Row  9301  no. of null values = 1 percentage =  4.76
Row  9303  no. of null values = 2 percentage =  9.52
Row  9304  no. of null values = 2 percentage =  9.52
Row  9305  no. of null values = 1 percentage =  4.76
Row  9309  no. of null values = 2 percentage =  9.52
Row  9310  no. of null values = 1 percentage =  4.76
Row  9311  no. of null values = 2 percentage =  9.52
Row  9312  no. of null values = 9 percentage =  42.86
Row  9313  no. of null values = 1 percentage =  4.76
Row  9314  no. of null values = 2 percentage

Row  9528  no. of null values = 8 percentage =  38.1
Row  9529  no. of null values = 2 percentage =  9.52
Row  9530  no. of null values = 7 percentage =  33.33
Row  9532  no. of null values = 7 percentage =  33.33
Row  9533  no. of null values = 3 percentage =  14.29
Row  9534  no. of null values = 2 percentage =  9.52
Row  9535  no. of null values = 3 percentage =  14.29
Row  9536  no. of null values = 7 percentage =  33.33
Row  9538  no. of null values = 9 percentage =  42.86
Row  9539  no. of null values = 7 percentage =  33.33
Row  9540  no. of null values = 2 percentage =  9.52
Row  9541  no. of null values = 9 percentage =  42.86
Row  9543  no. of null values = 9 percentage =  42.86
Row  9544  no. of null values = 7 percentage =  33.33
Row  9546  no. of null values = 9 percentage =  42.86
Row  9547  no. of null values = 10 percentage =  47.62
Row  9548  no. of null values = 9 percentage =  42.86
Row  9549  no. of null values = 8 percentage =  38.1
Row  9551  no. of null values = 

Row  9780  no. of null values = 2 percentage =  9.52
Row  9785  no. of null values = 9 percentage =  42.86
Row  9786  no. of null values = 9 percentage =  42.86
Row  9787  no. of null values = 9 percentage =  42.86
Row  9790  no. of null values = 9 percentage =  42.86
Row  9791  no. of null values = 3 percentage =  14.29
Row  9792  no. of null values = 9 percentage =  42.86
Row  9793  no. of null values = 2 percentage =  9.52
Row  9794  no. of null values = 2 percentage =  9.52
Row  9795  no. of null values = 3 percentage =  14.29
Row  9796  no. of null values = 2 percentage =  9.52
Row  9797  no. of null values = 2 percentage =  9.52
Row  9798  no. of null values = 10 percentage =  47.62
Row  9799  no. of null values = 2 percentage =  9.52
Row  9800  no. of null values = 2 percentage =  9.52
Row  9801  no. of null values = 2 percentage =  9.52
Row  9802  no. of null values = 1 percentage =  4.76
Row  9803  no. of null values = 7 percentage =  33.33
Row  9804  no. of null values = 2 pe

Row  10120  no. of null values = 2 percentage =  9.52
Row  10121  no. of null values = 10 percentage =  47.62
Row  10123  no. of null values = 2 percentage =  9.52
Row  10124  no. of null values = 2 percentage =  9.52
Row  10125  no. of null values = 21 percentage =  100.0
Row  10127  no. of null values = 3 percentage =  14.29
Row  10129  no. of null values = 1 percentage =  4.76
Row  10130  no. of null values = 1 percentage =  4.76
Row  10133  no. of null values = 9 percentage =  42.86
Row  10134  no. of null values = 2 percentage =  9.52
Row  10136  no. of null values = 9 percentage =  42.86
Row  10138  no. of null values = 1 percentage =  4.76
Row  10139  no. of null values = 2 percentage =  9.52
Row  10140  no. of null values = 3 percentage =  14.29
Row  10141  no. of null values = 1 percentage =  4.76
Row  10142  no. of null values = 10 percentage =  47.62
Row  10143  no. of null values = 1 percentage =  4.76
Row  10144  no. of null values = 2 percentage =  9.52
Row  10146  no. of

Row  10339  no. of null values = 1 percentage =  4.76
Row  10340  no. of null values = 2 percentage =  9.52
Row  10341  no. of null values = 1 percentage =  4.76
Row  10342  no. of null values = 9 percentage =  42.86
Row  10343  no. of null values = 1 percentage =  4.76
Row  10344  no. of null values = 10 percentage =  47.62
Row  10346  no. of null values = 2 percentage =  9.52
Row  10347  no. of null values = 10 percentage =  47.62
Row  10349  no. of null values = 2 percentage =  9.52
Row  10351  no. of null values = 2 percentage =  9.52
Row  10353  no. of null values = 9 percentage =  42.86
Row  10354  no. of null values = 2 percentage =  9.52
Row  10356  no. of null values = 2 percentage =  9.52
Row  10358  no. of null values = 9 percentage =  42.86
Row  10359  no. of null values = 2 percentage =  9.52
Row  10360  no. of null values = 9 percentage =  42.86
Row  10362  no. of null values = 9 percentage =  42.86
Row  10372  no. of null values = 2 percentage =  9.52
Row  10373  no. of 

Row  10571  no. of null values = 1 percentage =  4.76
Row  10572  no. of null values = 9 percentage =  42.86
Row  10573  no. of null values = 2 percentage =  9.52
Row  10575  no. of null values = 2 percentage =  9.52
Row  10576  no. of null values = 1 percentage =  4.76
Row  10577  no. of null values = 1 percentage =  4.76
Row  10578  no. of null values = 3 percentage =  14.29
Row  10579  no. of null values = 4 percentage =  19.05
Row  10580  no. of null values = 2 percentage =  9.52
Row  10583  no. of null values = 10 percentage =  47.62
Row  10584  no. of null values = 9 percentage =  42.86
Row  10586  no. of null values = 3 percentage =  14.29
Row  10588  no. of null values = 1 percentage =  4.76
Row  10589  no. of null values = 2 percentage =  9.52
Row  10590  no. of null values = 3 percentage =  14.29
Row  10591  no. of null values = 2 percentage =  9.52
Row  10592  no. of null values = 9 percentage =  42.86
Row  10593  no. of null values = 1 percentage =  4.76
Row  10594  no. of 

Row  10890  no. of null values = 9 percentage =  42.86
Row  10891  no. of null values = 2 percentage =  9.52
Row  10893  no. of null values = 9 percentage =  42.86
Row  10894  no. of null values = 9 percentage =  42.86
Row  10895  no. of null values = 9 percentage =  42.86
Row  10898  no. of null values = 10 percentage =  47.62
Row  10902  no. of null values = 1 percentage =  4.76
Row  10904  no. of null values = 2 percentage =  9.52
Row  10905  no. of null values = 2 percentage =  9.52
Row  10907  no. of null values = 9 percentage =  42.86
Row  10909  no. of null values = 9 percentage =  42.86
Row  10910  no. of null values = 9 percentage =  42.86
Row  10911  no. of null values = 1 percentage =  4.76
Row  10912  no. of null values = 2 percentage =  9.52
Row  10913  no. of null values = 1 percentage =  4.76
Row  10915  no. of null values = 9 percentage =  42.86
Row  10916  no. of null values = 2 percentage =  9.52
Row  10918  no. of null values = 2 percentage =  9.52
Row  10920  no. of

Row  11257  no. of null values = 1 percentage =  4.76
Row  11258  no. of null values = 9 percentage =  42.86
Row  11260  no. of null values = 9 percentage =  42.86
Row  11261  no. of null values = 2 percentage =  9.52
Row  11262  no. of null values = 2 percentage =  9.52
Row  11263  no. of null values = 2 percentage =  9.52
Row  11264  no. of null values = 8 percentage =  38.1
Row  11266  no. of null values = 2 percentage =  9.52
Row  11267  no. of null values = 2 percentage =  9.52
Row  11268  no. of null values = 9 percentage =  42.86
Row  11269  no. of null values = 9 percentage =  42.86
Row  11270  no. of null values = 2 percentage =  9.52
Row  11272  no. of null values = 3 percentage =  14.29
Row  11273  no. of null values = 2 percentage =  9.52
Row  11274  no. of null values = 9 percentage =  42.86
Row  11275  no. of null values = 2 percentage =  9.52
Row  11276  no. of null values = 3 percentage =  14.29
Row  11278  no. of null values = 1 percentage =  4.76
Row  11279  no. of nu

Row  11546  no. of null values = 2 percentage =  9.52
Row  11547  no. of null values = 10 percentage =  47.62
Row  11550  no. of null values = 9 percentage =  42.86
Row  11551  no. of null values = 3 percentage =  14.29
Row  11552  no. of null values = 21 percentage =  100.0
Row  11554  no. of null values = 1 percentage =  4.76
Row  11555  no. of null values = 2 percentage =  9.52
Row  11556  no. of null values = 2 percentage =  9.52
Row  11557  no. of null values = 2 percentage =  9.52
Row  11558  no. of null values = 10 percentage =  47.62
Row  11559  no. of null values = 3 percentage =  14.29
Row  11560  no. of null values = 2 percentage =  9.52
Row  11561  no. of null values = 2 percentage =  9.52
Row  11563  no. of null values = 10 percentage =  47.62
Row  11564  no. of null values = 2 percentage =  9.52
Row  11565  no. of null values = 2 percentage =  9.52
Row  11566  no. of null values = 1 percentage =  4.76
Row  11567  no. of null values = 1 percentage =  4.76
Row  11568  no. o

Row  11922  no. of null values = 2 percentage =  9.52
Row  11926  no. of null values = 2 percentage =  9.52
Row  11928  no. of null values = 2 percentage =  9.52
Row  11930  no. of null values = 2 percentage =  9.52
Row  11931  no. of null values = 1 percentage =  4.76
Row  11932  no. of null values = 9 percentage =  42.86
Row  11933  no. of null values = 2 percentage =  9.52
Row  11934  no. of null values = 2 percentage =  9.52
Row  11936  no. of null values = 2 percentage =  9.52
Row  11937  no. of null values = 1 percentage =  4.76
Row  11938  no. of null values = 2 percentage =  9.52
Row  11939  no. of null values = 9 percentage =  42.86
Row  11940  no. of null values = 9 percentage =  42.86
Row  11941  no. of null values = 9 percentage =  42.86
Row  11942  no. of null values = 9 percentage =  42.86
Row  11943  no. of null values = 10 percentage =  47.62
Row  11944  no. of null values = 10 percentage =  47.62
Row  11945  no. of null values = 2 percentage =  9.52
Row  11946  no. of 

Row  12207  no. of null values = 9 percentage =  42.86
Row  12208  no. of null values = 9 percentage =  42.86
Row  12209  no. of null values = 10 percentage =  47.62
Row  12210  no. of null values = 10 percentage =  47.62
Row  12212  no. of null values = 2 percentage =  9.52
Row  12213  no. of null values = 2 percentage =  9.52
Row  12214  no. of null values = 2 percentage =  9.52
Row  12215  no. of null values = 3 percentage =  14.29
Row  12217  no. of null values = 9 percentage =  42.86
Row  12218  no. of null values = 3 percentage =  14.29
Row  12219  no. of null values = 10 percentage =  47.62
Row  12220  no. of null values = 1 percentage =  4.76
Row  12221  no. of null values = 10 percentage =  47.62
Row  12222  no. of null values = 1 percentage =  4.76
Row  12223  no. of null values = 2 percentage =  9.52
Row  12224  no. of null values = 2 percentage =  9.52
Row  12226  no. of null values = 2 percentage =  9.52
Row  12227  no. of null values = 1 percentage =  4.76
Row  12228  no.

Row  12482  no. of null values = 2 percentage =  9.52
Row  12483  no. of null values = 1 percentage =  4.76
Row  12484  no. of null values = 2 percentage =  9.52
Row  12486  no. of null values = 9 percentage =  42.86
Row  12487  no. of null values = 2 percentage =  9.52
Row  12489  no. of null values = 2 percentage =  9.52
Row  12490  no. of null values = 9 percentage =  42.86
Row  12491  no. of null values = 2 percentage =  9.52
Row  12492  no. of null values = 1 percentage =  4.76
Row  12493  no. of null values = 9 percentage =  42.86
Row  12495  no. of null values = 10 percentage =  47.62
Row  12496  no. of null values = 1 percentage =  4.76
Row  12497  no. of null values = 2 percentage =  9.52
Row  12499  no. of null values = 2 percentage =  9.52
Row  12501  no. of null values = 1 percentage =  4.76
Row  12502  no. of null values = 2 percentage =  9.52
Row  12503  no. of null values = 2 percentage =  9.52
Row  12504  no. of null values = 2 percentage =  9.52
Row  12506  no. of null

Row  12768  no. of null values = 3 percentage =  14.29
Row  12769  no. of null values = 9 percentage =  42.86
Row  12770  no. of null values = 1 percentage =  4.76
Row  12771  no. of null values = 9 percentage =  42.86
Row  12772  no. of null values = 1 percentage =  4.76
Row  12775  no. of null values = 2 percentage =  9.52
Row  12776  no. of null values = 10 percentage =  47.62
Row  12777  no. of null values = 10 percentage =  47.62
Row  12778  no. of null values = 9 percentage =  42.86
Row  12779  no. of null values = 9 percentage =  42.86
Row  12780  no. of null values = 9 percentage =  42.86
Row  12781  no. of null values = 9 percentage =  42.86
Row  12783  no. of null values = 9 percentage =  42.86
Row  12784  no. of null values = 10 percentage =  47.62
Row  12785  no. of null values = 1 percentage =  4.76
Row  12786  no. of null values = 1 percentage =  4.76
Row  12787  no. of null values = 1 percentage =  4.76
Row  12788  no. of null values = 2 percentage =  9.52
Row  12790  no

Row  13084  no. of null values = 9 percentage =  42.86
Row  13085  no. of null values = 2 percentage =  9.52
Row  13088  no. of null values = 1 percentage =  4.76
Row  13090  no. of null values = 2 percentage =  9.52
Row  13091  no. of null values = 9 percentage =  42.86
Row  13092  no. of null values = 2 percentage =  9.52
Row  13093  no. of null values = 1 percentage =  4.76
Row  13094  no. of null values = 2 percentage =  9.52
Row  13095  no. of null values = 2 percentage =  9.52
Row  13096  no. of null values = 9 percentage =  42.86
Row  13097  no. of null values = 2 percentage =  9.52
Row  13099  no. of null values = 1 percentage =  4.76
Row  13100  no. of null values = 10 percentage =  47.62
Row  13102  no. of null values = 2 percentage =  9.52
Row  13103  no. of null values = 2 percentage =  9.52
Row  13106  no. of null values = 1 percentage =  4.76
Row  13109  no. of null values = 2 percentage =  9.52
Row  13110  no. of null values = 2 percentage =  9.52
Row  13111  no. of null

Row  13352  no. of null values = 9 percentage =  42.86
Row  13353  no. of null values = 1 percentage =  4.76
Row  13354  no. of null values = 2 percentage =  9.52
Row  13355  no. of null values = 9 percentage =  42.86
Row  13356  no. of null values = 3 percentage =  14.29
Row  13358  no. of null values = 2 percentage =  9.52
Row  13359  no. of null values = 1 percentage =  4.76
Row  13361  no. of null values = 9 percentage =  42.86
Row  13363  no. of null values = 1 percentage =  4.76
Row  13366  no. of null values = 2 percentage =  9.52
Row  13367  no. of null values = 3 percentage =  14.29
Row  13368  no. of null values = 2 percentage =  9.52
Row  13369  no. of null values = 2 percentage =  9.52
Row  13370  no. of null values = 2 percentage =  9.52
Row  13371  no. of null values = 2 percentage =  9.52
Row  13372  no. of null values = 2 percentage =  9.52
Row  13373  no. of null values = 1 percentage =  4.76
Row  13374  no. of null values = 9 percentage =  42.86
Row  13375  no. of nul

Row  13591  no. of null values = 2 percentage =  9.52
Row  13592  no. of null values = 9 percentage =  42.86
Row  13593  no. of null values = 9 percentage =  42.86
Row  13594  no. of null values = 1 percentage =  4.76
Row  13595  no. of null values = 1 percentage =  4.76
Row  13596  no. of null values = 9 percentage =  42.86
Row  13598  no. of null values = 2 percentage =  9.52
Row  13599  no. of null values = 9 percentage =  42.86
Row  13600  no. of null values = 1 percentage =  4.76
Row  13601  no. of null values = 2 percentage =  9.52
Row  13603  no. of null values = 2 percentage =  9.52
Row  13604  no. of null values = 2 percentage =  9.52
Row  13605  no. of null values = 1 percentage =  4.76
Row  13606  no. of null values = 9 percentage =  42.86
Row  13607  no. of null values = 2 percentage =  9.52
Row  13608  no. of null values = 9 percentage =  42.86
Row  13609  no. of null values = 2 percentage =  9.52
Row  13611  no. of null values = 9 percentage =  42.86
Row  13612  no. of nu

Row  13889  no. of null values = 9 percentage =  42.86
Row  13890  no. of null values = 3 percentage =  14.29
Row  13891  no. of null values = 1 percentage =  4.76
Row  13892  no. of null values = 3 percentage =  14.29
Row  13894  no. of null values = 1 percentage =  4.76
Row  13895  no. of null values = 2 percentage =  9.52
Row  13896  no. of null values = 2 percentage =  9.52
Row  13897  no. of null values = 2 percentage =  9.52
Row  13898  no. of null values = 2 percentage =  9.52
Row  13899  no. of null values = 2 percentage =  9.52
Row  13900  no. of null values = 9 percentage =  42.86
Row  13901  no. of null values = 9 percentage =  42.86
Row  13902  no. of null values = 10 percentage =  47.62
Row  13903  no. of null values = 2 percentage =  9.52
Row  13904  no. of null values = 9 percentage =  42.86
Row  13905  no. of null values = 1 percentage =  4.76
Row  13906  no. of null values = 9 percentage =  42.86
Row  13907  no. of null values = 2 percentage =  9.52
Row  13908  no. of 

Row  14149  no. of null values = 1 percentage =  4.76
Row  14150  no. of null values = 1 percentage =  4.76
Row  14151  no. of null values = 9 percentage =  42.86
Row  14152  no. of null values = 3 percentage =  14.29
Row  14153  no. of null values = 1 percentage =  4.76
Row  14155  no. of null values = 2 percentage =  9.52
Row  14156  no. of null values = 10 percentage =  47.62
Row  14157  no. of null values = 1 percentage =  4.76
Row  14158  no. of null values = 9 percentage =  42.86
Row  14159  no. of null values = 9 percentage =  42.86
Row  14160  no. of null values = 1 percentage =  4.76
Row  14161  no. of null values = 1 percentage =  4.76
Row  14162  no. of null values = 1 percentage =  4.76
Row  14163  no. of null values = 9 percentage =  42.86
Row  14164  no. of null values = 2 percentage =  9.52
Row  14165  no. of null values = 10 percentage =  47.62
Row  14167  no. of null values = 2 percentage =  9.52
Row  14171  no. of null values = 9 percentage =  42.86
Row  14172  no. of

Row  14358  no. of null values = 2 percentage =  9.52
Row  14362  no. of null values = 9 percentage =  42.86
Row  14363  no. of null values = 2 percentage =  9.52
Row  14364  no. of null values = 2 percentage =  9.52
Row  14365  no. of null values = 3 percentage =  14.29
Row  14366  no. of null values = 2 percentage =  9.52
Row  14367  no. of null values = 1 percentage =  4.76
Row  14368  no. of null values = 2 percentage =  9.52
Row  14369  no. of null values = 3 percentage =  14.29
Row  14370  no. of null values = 2 percentage =  9.52
Row  14371  no. of null values = 2 percentage =  9.52
Row  14372  no. of null values = 9 percentage =  42.86
Row  14374  no. of null values = 1 percentage =  4.76
Row  14375  no. of null values = 2 percentage =  9.52
Row  14378  no. of null values = 10 percentage =  47.62
Row  14379  no. of null values = 2 percentage =  9.52
Row  14383  no. of null values = 1 percentage =  4.76
Row  14384  no. of null values = 2 percentage =  9.52
Row  14385  no. of nul

Row  14734  no. of null values = 1 percentage =  4.76
Row  14735  no. of null values = 2 percentage =  9.52
Row  14736  no. of null values = 2 percentage =  9.52
Row  14737  no. of null values = 2 percentage =  9.52
Row  14738  no. of null values = 9 percentage =  42.86
Row  14740  no. of null values = 1 percentage =  4.76
Row  14741  no. of null values = 9 percentage =  42.86
Row  14742  no. of null values = 10 percentage =  47.62
Row  14743  no. of null values = 9 percentage =  42.86
Row  14744  no. of null values = 2 percentage =  9.52
Row  14747  no. of null values = 9 percentage =  42.86
Row  14748  no. of null values = 2 percentage =  9.52
Row  14749  no. of null values = 3 percentage =  14.29
Row  14750  no. of null values = 9 percentage =  42.86
Row  14751  no. of null values = 3 percentage =  14.29
Row  14752  no. of null values = 3 percentage =  14.29
Row  14754  no. of null values = 2 percentage =  9.52
Row  14755  no. of null values = 3 percentage =  14.29
Row  14756  no. o

Row  15001  no. of null values = 10 percentage =  47.62
Row  15002  no. of null values = 3 percentage =  14.29
Row  15004  no. of null values = 1 percentage =  4.76
Row  15005  no. of null values = 9 percentage =  42.86
Row  15006  no. of null values = 2 percentage =  9.52
Row  15007  no. of null values = 1 percentage =  4.76
Row  15008  no. of null values = 3 percentage =  14.29
Row  15009  no. of null values = 9 percentage =  42.86
Row  15010  no. of null values = 1 percentage =  4.76
Row  15011  no. of null values = 3 percentage =  14.29
Row  15012  no. of null values = 9 percentage =  42.86
Row  15016  no. of null values = 1 percentage =  4.76
Row  15017  no. of null values = 9 percentage =  42.86
Row  15019  no. of null values = 1 percentage =  4.76
Row  15021  no. of null values = 2 percentage =  9.52
Row  15024  no. of null values = 2 percentage =  9.52
Row  15025  no. of null values = 9 percentage =  42.86
Row  15026  no. of null values = 2 percentage =  9.52
Row  15028  no. of

Row  15337  no. of null values = 1 percentage =  4.76
Row  15338  no. of null values = 2 percentage =  9.52
Row  15340  no. of null values = 9 percentage =  42.86
Row  15343  no. of null values = 9 percentage =  42.86
Row  15344  no. of null values = 2 percentage =  9.52
Row  15345  no. of null values = 2 percentage =  9.52
Row  15346  no. of null values = 1 percentage =  4.76
Row  15348  no. of null values = 10 percentage =  47.62
Row  15349  no. of null values = 2 percentage =  9.52
Row  15352  no. of null values = 9 percentage =  42.86
Row  15353  no. of null values = 9 percentage =  42.86
Row  15354  no. of null values = 10 percentage =  47.62
Row  15355  no. of null values = 1 percentage =  4.76
Row  15356  no. of null values = 1 percentage =  4.76
Row  15357  no. of null values = 2 percentage =  9.52
Row  15358  no. of null values = 1 percentage =  4.76
Row  15359  no. of null values = 2 percentage =  9.52
Row  15363  no. of null values = 9 percentage =  42.86
Row  15365  no. of 

Row  15658  no. of null values = 2 percentage =  9.52
Row  15659  no. of null values = 9 percentage =  42.86
Row  15661  no. of null values = 2 percentage =  9.52
Row  15662  no. of null values = 2 percentage =  9.52
Row  15663  no. of null values = 3 percentage =  14.29
Row  15665  no. of null values = 2 percentage =  9.52
Row  15666  no. of null values = 9 percentage =  42.86
Row  15667  no. of null values = 9 percentage =  42.86
Row  15668  no. of null values = 2 percentage =  9.52
Row  15669  no. of null values = 9 percentage =  42.86
Row  15670  no. of null values = 2 percentage =  9.52
Row  15672  no. of null values = 9 percentage =  42.86
Row  15673  no. of null values = 1 percentage =  4.76
Row  15675  no. of null values = 9 percentage =  42.86
Row  15676  no. of null values = 1 percentage =  4.76
Row  15677  no. of null values = 2 percentage =  9.52
Row  15682  no. of null values = 2 percentage =  9.52
Row  15683  no. of null values = 2 percentage =  9.52
Row  15685  no. of nu

Row  15940  no. of null values = 2 percentage =  9.52
Row  15941  no. of null values = 2 percentage =  9.52
Row  15942  no. of null values = 1 percentage =  4.76
Row  15943  no. of null values = 2 percentage =  9.52
Row  15944  no. of null values = 1 percentage =  4.76
Row  15945  no. of null values = 9 percentage =  42.86
Row  15946  no. of null values = 10 percentage =  47.62
Row  15948  no. of null values = 1 percentage =  4.76
Row  15949  no. of null values = 1 percentage =  4.76
Row  15950  no. of null values = 1 percentage =  4.76
Row  15951  no. of null values = 2 percentage =  9.52
Row  15953  no. of null values = 2 percentage =  9.52
Row  15955  no. of null values = 1 percentage =  4.76
Row  15960  no. of null values = 9 percentage =  42.86
Row  15961  no. of null values = 9 percentage =  42.86
Row  15965  no. of null values = 2 percentage =  9.52
Row  15966  no. of null values = 1 percentage =  4.76
Row  15967  no. of null values = 2 percentage =  9.52
Row  15968  no. of null

Row  16223  no. of null values = 2 percentage =  9.52
Row  16224  no. of null values = 10 percentage =  47.62
Row  16225  no. of null values = 9 percentage =  42.86
Row  16226  no. of null values = 2 percentage =  9.52
Row  16227  no. of null values = 2 percentage =  9.52
Row  16228  no. of null values = 2 percentage =  9.52
Row  16229  no. of null values = 1 percentage =  4.76
Row  16230  no. of null values = 2 percentage =  9.52
Row  16231  no. of null values = 1 percentage =  4.76
Row  16232  no. of null values = 9 percentage =  42.86
Row  16233  no. of null values = 2 percentage =  9.52
Row  16234  no. of null values = 3 percentage =  14.29
Row  16235  no. of null values = 9 percentage =  42.86
Row  16236  no. of null values = 1 percentage =  4.76
Row  16237  no. of null values = 2 percentage =  9.52
Row  16238  no. of null values = 9 percentage =  42.86
Row  16239  no. of null values = 3 percentage =  14.29
Row  16241  no. of null values = 2 percentage =  9.52
Row  16242  no. of n

Row  16471  no. of null values = 1 percentage =  4.76
Row  16472  no. of null values = 9 percentage =  42.86
Row  16473  no. of null values = 10 percentage =  47.62
Row  16474  no. of null values = 3 percentage =  14.29
Row  16475  no. of null values = 10 percentage =  47.62
Row  16476  no. of null values = 10 percentage =  47.62
Row  16477  no. of null values = 2 percentage =  9.52
Row  16478  no. of null values = 1 percentage =  4.76
Row  16479  no. of null values = 1 percentage =  4.76
Row  16480  no. of null values = 1 percentage =  4.76
Row  16482  no. of null values = 10 percentage =  47.62
Row  16484  no. of null values = 2 percentage =  9.52
Row  16485  no. of null values = 9 percentage =  42.86
Row  16486  no. of null values = 7 percentage =  33.33
Row  16487  no. of null values = 9 percentage =  42.86
Row  16490  no. of null values = 2 percentage =  9.52
Row  16491  no. of null values = 1 percentage =  4.76
Row  16492  no. of null values = 1 percentage =  4.76
Row  16493  no.

Row  16703  no. of null values = 1 percentage =  4.76
Row  16705  no. of null values = 2 percentage =  9.52
Row  16706  no. of null values = 2 percentage =  9.52
Row  16707  no. of null values = 2 percentage =  9.52
Row  16709  no. of null values = 9 percentage =  42.86
Row  16710  no. of null values = 2 percentage =  9.52
Row  16711  no. of null values = 3 percentage =  14.29
Row  16712  no. of null values = 9 percentage =  42.86
Row  16713  no. of null values = 3 percentage =  14.29
Row  16714  no. of null values = 1 percentage =  4.76
Row  16715  no. of null values = 9 percentage =  42.86
Row  16716  no. of null values = 9 percentage =  42.86
Row  16717  no. of null values = 2 percentage =  9.52
Row  16719  no. of null values = 2 percentage =  9.52
Row  16720  no. of null values = 9 percentage =  42.86
Row  16721  no. of null values = 2 percentage =  9.52
Row  16722  no. of null values = 2 percentage =  9.52
Row  16723  no. of null values = 1 percentage =  4.76
Row  16724  no. of nu

Row  16949  no. of null values = 2 percentage =  9.52
Row  16950  no. of null values = 2 percentage =  9.52
Row  16951  no. of null values = 2 percentage =  9.52
Row  16953  no. of null values = 2 percentage =  9.52
Row  16955  no. of null values = 1 percentage =  4.76
Row  16956  no. of null values = 2 percentage =  9.52
Row  16957  no. of null values = 10 percentage =  47.62
Row  16958  no. of null values = 9 percentage =  42.86
Row  16959  no. of null values = 2 percentage =  9.52
Row  16960  no. of null values = 2 percentage =  9.52
Row  16962  no. of null values = 9 percentage =  42.86
Row  16966  no. of null values = 9 percentage =  42.86
Row  16967  no. of null values = 9 percentage =  42.86
Row  16968  no. of null values = 1 percentage =  4.76
Row  16969  no. of null values = 9 percentage =  42.86
Row  16970  no. of null values = 1 percentage =  4.76
Row  16971  no. of null values = 9 percentage =  42.86
Row  16972  no. of null values = 2 percentage =  9.52
Row  16973  no. of n

Row  17273  no. of null values = 2 percentage =  9.52
Row  17274  no. of null values = 10 percentage =  47.62
Row  17277  no. of null values = 9 percentage =  42.86
Row  17279  no. of null values = 2 percentage =  9.52
Row  17281  no. of null values = 2 percentage =  9.52
Row  17284  no. of null values = 9 percentage =  42.86
Row  17286  no. of null values = 3 percentage =  14.29
Row  17287  no. of null values = 2 percentage =  9.52
Row  17288  no. of null values = 9 percentage =  42.86
Row  17289  no. of null values = 1 percentage =  4.76
Row  17290  no. of null values = 1 percentage =  4.76
Row  17291  no. of null values = 9 percentage =  42.86
Row  17292  no. of null values = 2 percentage =  9.52
Row  17293  no. of null values = 1 percentage =  4.76
Row  17294  no. of null values = 2 percentage =  9.52
Row  17296  no. of null values = 2 percentage =  9.52
Row  17297  no. of null values = 1 percentage =  4.76
Row  17298  no. of null values = 9 percentage =  42.86
Row  17299  no. of n

Row  17590  no. of null values = 9 percentage =  42.86
Row  17591  no. of null values = 2 percentage =  9.52
Row  17594  no. of null values = 3 percentage =  14.29
Row  17595  no. of null values = 5 percentage =  23.81
Row  17596  no. of null values = 2 percentage =  9.52
Row  17597  no. of null values = 1 percentage =  4.76
Row  17598  no. of null values = 2 percentage =  9.52
Row  17602  no. of null values = 10 percentage =  47.62
Row  17603  no. of null values = 2 percentage =  9.52
Row  17605  no. of null values = 2 percentage =  9.52
Row  17606  no. of null values = 2 percentage =  9.52
Row  17607  no. of null values = 1 percentage =  4.76
Row  17608  no. of null values = 1 percentage =  4.76
Row  17611  no. of null values = 21 percentage =  100.0
Row  17612  no. of null values = 9 percentage =  42.86
Row  17614  no. of null values = 2 percentage =  9.52
Row  17616  no. of null values = 2 percentage =  9.52
Row  17617  no. of null values = 3 percentage =  14.29
Row  17618  no. of 

Row  17947  no. of null values = 2 percentage =  9.52
Row  17948  no. of null values = 1 percentage =  4.76
Row  17950  no. of null values = 2 percentage =  9.52
Row  17951  no. of null values = 9 percentage =  42.86
Row  17954  no. of null values = 9 percentage =  42.86
Row  17955  no. of null values = 1 percentage =  4.76
Row  17956  no. of null values = 2 percentage =  9.52
Row  17958  no. of null values = 10 percentage =  47.62
Row  17959  no. of null values = 2 percentage =  9.52
Row  17960  no. of null values = 10 percentage =  47.62
Row  17963  no. of null values = 2 percentage =  9.52
Row  17964  no. of null values = 3 percentage =  14.29
Row  17965  no. of null values = 9 percentage =  42.86
Row  17966  no. of null values = 2 percentage =  9.52
Row  17967  no. of null values = 10 percentage =  47.62
Row  17968  no. of null values = 2 percentage =  9.52
Row  17970  no. of null values = 2 percentage =  9.52
Row  17971  no. of null values = 9 percentage =  42.86
Row  17973  no. o

Row  18391  no. of null values = 1 percentage =  4.76
Row  18392  no. of null values = 3 percentage =  14.29
Row  18393  no. of null values = 1 percentage =  4.76
Row  18394  no. of null values = 1 percentage =  4.76
Row  18397  no. of null values = 9 percentage =  42.86
Row  18398  no. of null values = 1 percentage =  4.76
Row  18400  no. of null values = 2 percentage =  9.52
Row  18401  no. of null values = 2 percentage =  9.52
Row  18402  no. of null values = 1 percentage =  4.76
Row  18403  no. of null values = 1 percentage =  4.76
Row  18404  no. of null values = 2 percentage =  9.52
Row  18405  no. of null values = 2 percentage =  9.52
Row  18406  no. of null values = 2 percentage =  9.52
Row  18407  no. of null values = 3 percentage =  14.29
Row  18408  no. of null values = 3 percentage =  14.29
Row  18409  no. of null values = 2 percentage =  9.52
Row  18410  no. of null values = 2 percentage =  9.52
Row  18413  no. of null values = 8 percentage =  38.1
Row  18416  no. of null 

Row  18655  no. of null values = 2 percentage =  9.52
Row  18656  no. of null values = 2 percentage =  9.52
Row  18657  no. of null values = 9 percentage =  42.86
Row  18658  no. of null values = 2 percentage =  9.52
Row  18659  no. of null values = 1 percentage =  4.76
Row  18660  no. of null values = 1 percentage =  4.76
Row  18661  no. of null values = 2 percentage =  9.52
Row  18664  no. of null values = 1 percentage =  4.76
Row  18665  no. of null values = 9 percentage =  42.86
Row  18666  no. of null values = 1 percentage =  4.76
Row  18668  no. of null values = 1 percentage =  4.76
Row  18671  no. of null values = 21 percentage =  100.0
Row  18672  no. of null values = 2 percentage =  9.52
Row  18673  no. of null values = 1 percentage =  4.76
Row  18674  no. of null values = 1 percentage =  4.76
Row  18676  no. of null values = 2 percentage =  9.52
Row  18677  no. of null values = 1 percentage =  4.76
Row  18678  no. of null values = 1 percentage =  4.76
Row  18679  no. of null 

Row  18896  no. of null values = 2 percentage =  9.52
Row  18897  no. of null values = 2 percentage =  9.52
Row  18901  no. of null values = 2 percentage =  9.52
Row  18903  no. of null values = 2 percentage =  9.52
Row  18904  no. of null values = 1 percentage =  4.76
Row  18905  no. of null values = 9 percentage =  42.86
Row  18906  no. of null values = 1 percentage =  4.76
Row  18907  no. of null values = 3 percentage =  14.29
Row  18909  no. of null values = 1 percentage =  4.76
Row  18910  no. of null values = 2 percentage =  9.52
Row  18912  no. of null values = 2 percentage =  9.52
Row  18913  no. of null values = 2 percentage =  9.52
Row  18915  no. of null values = 2 percentage =  9.52
Row  18916  no. of null values = 1 percentage =  4.76
Row  18918  no. of null values = 1 percentage =  4.76
Row  18919  no. of null values = 1 percentage =  4.76
Row  18920  no. of null values = 2 percentage =  9.52
Row  18921  no. of null values = 2 percentage =  9.52
Row  18922  no. of null va

Row  19160  no. of null values = 2 percentage =  9.52
Row  19161  no. of null values = 3 percentage =  14.29
Row  19163  no. of null values = 9 percentage =  42.86
Row  19164  no. of null values = 2 percentage =  9.52
Row  19166  no. of null values = 9 percentage =  42.86
Row  19167  no. of null values = 2 percentage =  9.52
Row  19168  no. of null values = 2 percentage =  9.52
Row  19169  no. of null values = 1 percentage =  4.76
Row  19171  no. of null values = 2 percentage =  9.52
Row  19172  no. of null values = 2 percentage =  9.52
Row  19173  no. of null values = 1 percentage =  4.76
Row  19174  no. of null values = 2 percentage =  9.52
Row  19175  no. of null values = 1 percentage =  4.76
Row  19176  no. of null values = 10 percentage =  47.62
Row  19177  no. of null values = 9 percentage =  42.86
Row  19178  no. of null values = 9 percentage =  42.86
Row  19179  no. of null values = 1 percentage =  4.76
Row  19183  no. of null values = 1 percentage =  4.76
Row  19184  no. of nu

Row  19516  no. of null values = 2 percentage =  9.52
Row  19517  no. of null values = 2 percentage =  9.52
Row  19518  no. of null values = 1 percentage =  4.76
Row  19519  no. of null values = 1 percentage =  4.76
Row  19521  no. of null values = 3 percentage =  14.29
Row  19522  no. of null values = 2 percentage =  9.52
Row  19523  no. of null values = 2 percentage =  9.52
Row  19524  no. of null values = 2 percentage =  9.52
Row  19526  no. of null values = 9 percentage =  42.86
Row  19527  no. of null values = 2 percentage =  9.52
Row  19528  no. of null values = 10 percentage =  47.62
Row  19529  no. of null values = 2 percentage =  9.52
Row  19530  no. of null values = 9 percentage =  42.86
Row  19531  no. of null values = 9 percentage =  42.86
Row  19533  no. of null values = 2 percentage =  9.52
Row  19535  no. of null values = 2 percentage =  9.52
Row  19536  no. of null values = 1 percentage =  4.76
Row  19539  no. of null values = 1 percentage =  4.76
Row  19540  no. of nul

Row  19753  no. of null values = 1 percentage =  4.76
Row  19754  no. of null values = 2 percentage =  9.52
Row  19756  no. of null values = 2 percentage =  9.52
Row  19757  no. of null values = 1 percentage =  4.76
Row  19758  no. of null values = 1 percentage =  4.76
Row  19760  no. of null values = 1 percentage =  4.76
Row  19761  no. of null values = 9 percentage =  42.86
Row  19762  no. of null values = 1 percentage =  4.76
Row  19763  no. of null values = 2 percentage =  9.52
Row  19764  no. of null values = 10 percentage =  47.62
Row  19766  no. of null values = 2 percentage =  9.52
Row  19767  no. of null values = 2 percentage =  9.52
Row  19769  no. of null values = 2 percentage =  9.52
Row  19774  no. of null values = 2 percentage =  9.52
Row  19777  no. of null values = 2 percentage =  9.52
Row  19780  no. of null values = 1 percentage =  4.76
Row  19782  no. of null values = 1 percentage =  4.76
Row  19783  no. of null values = 9 percentage =  42.86
Row  19784  no. of null 

Row  19979  no. of null values = 4 percentage =  19.05
Row  19980  no. of null values = 2 percentage =  9.52
Row  19981  no. of null values = 5 percentage =  23.81
Row  19982  no. of null values = 4 percentage =  19.05
Row  19983  no. of null values = 4 percentage =  19.05
Row  19984  no. of null values = 4 percentage =  19.05
Row  19985  no. of null values = 4 percentage =  19.05
Row  19986  no. of null values = 1 percentage =  4.76
Row  19987  no. of null values = 2 percentage =  9.52
Row  19988  no. of null values = 1 percentage =  4.76
Row  19989  no. of null values = 2 percentage =  9.52
Row  19990  no. of null values = 4 percentage =  19.05
Row  19991  no. of null values = 3 percentage =  14.29
Row  19992  no. of null values = 3 percentage =  14.29
Row  19993  no. of null values = 2 percentage =  9.52
Row  19994  no. of null values = 2 percentage =  9.52
Row  19995  no. of null values = 1 percentage =  4.76
Row  19996  no. of null values = 1 percentage =  4.76
Row  19997  no. of 

Row  20354  no. of null values = 1 percentage =  4.76
Row  20355  no. of null values = 1 percentage =  4.76
Row  20356  no. of null values = 1 percentage =  4.76
Row  20357  no. of null values = 3 percentage =  14.29
Row  20358  no. of null values = 3 percentage =  14.29
Row  20359  no. of null values = 3 percentage =  14.29
Row  20360  no. of null values = 3 percentage =  14.29
Row  20361  no. of null values = 2 percentage =  9.52
Row  20362  no. of null values = 1 percentage =  4.76
Row  20363  no. of null values = 3 percentage =  14.29
Row  20364  no. of null values = 4 percentage =  19.05
Row  20365  no. of null values = 2 percentage =  9.52
Row  20366  no. of null values = 3 percentage =  14.29
Row  20367  no. of null values = 2 percentage =  9.52
Row  20368  no. of null values = 1 percentage =  4.76
Row  20369  no. of null values = 4 percentage =  19.05
Row  20370  no. of null values = 2 percentage =  9.52
Row  20371  no. of null values = 4 percentage =  19.05
Row  20372  no. of 

Row  20552  no. of null values = 1 percentage =  4.76
Row  20553  no. of null values = 2 percentage =  9.52
Row  20554  no. of null values = 4 percentage =  19.05
Row  20555  no. of null values = 4 percentage =  19.05
Row  20556  no. of null values = 4 percentage =  19.05
Row  20557  no. of null values = 10 percentage =  47.62
Row  20558  no. of null values = 4 percentage =  19.05
Row  20559  no. of null values = 4 percentage =  19.05
Row  20560  no. of null values = 3 percentage =  14.29
Row  20561  no. of null values = 9 percentage =  42.86
Row  20562  no. of null values = 3 percentage =  14.29
Row  20563  no. of null values = 2 percentage =  9.52
Row  20564  no. of null values = 2 percentage =  9.52
Row  20565  no. of null values = 4 percentage =  19.05
Row  20566  no. of null values = 10 percentage =  47.62
Row  20567  no. of null values = 2 percentage =  9.52
Row  20568  no. of null values = 2 percentage =  9.52
Row  20569  no. of null values = 3 percentage =  14.29
Row  20570  no

Row  20729  no. of null values = 1 percentage =  4.76
Row  20730  no. of null values = 4 percentage =  19.05
Row  20731  no. of null values = 2 percentage =  9.52
Row  20732  no. of null values = 4 percentage =  19.05
Row  20733  no. of null values = 5 percentage =  23.81
Row  20734  no. of null values = 4 percentage =  19.05
Row  20735  no. of null values = 3 percentage =  14.29
Row  20736  no. of null values = 1 percentage =  4.76
Row  20737  no. of null values = 1 percentage =  4.76
Row  20738  no. of null values = 4 percentage =  19.05
Row  20739  no. of null values = 1 percentage =  4.76
Row  20740  no. of null values = 3 percentage =  14.29
Row  20741  no. of null values = 1 percentage =  4.76
Row  20742  no. of null values = 3 percentage =  14.29
Row  20743  no. of null values = 3 percentage =  14.29
Row  20744  no. of null values = 3 percentage =  14.29
Row  20745  no. of null values = 2 percentage =  9.52
Row  20746  no. of null values = 2 percentage =  9.52
Row  20747  no. of

Row  20954  no. of null values = 4 percentage =  19.05
Row  20955  no. of null values = 4 percentage =  19.05
Row  20956  no. of null values = 4 percentage =  19.05
Row  20957  no. of null values = 2 percentage =  9.52
Row  20958  no. of null values = 4 percentage =  19.05
Row  20959  no. of null values = 4 percentage =  19.05
Row  20960  no. of null values = 3 percentage =  14.29
Row  20961  no. of null values = 2 percentage =  9.52
Row  20962  no. of null values = 6 percentage =  28.57
Row  20963  no. of null values = 9 percentage =  42.86
Row  20964  no. of null values = 1 percentage =  4.76
Row  20965  no. of null values = 9 percentage =  42.86
Row  20966  no. of null values = 5 percentage =  23.81
Row  20967  no. of null values = 4 percentage =  19.05
Row  20968  no. of null values = 5 percentage =  23.81
Row  20969  no. of null values = 1 percentage =  4.76
Row  20970  no. of null values = 4 percentage =  19.05
Row  20971  no. of null values = 4 percentage =  19.05
Row  20972  no

Row  21145  no. of null values = 3 percentage =  14.29
Row  21146  no. of null values = 2 percentage =  9.52
Row  21147  no. of null values = 3 percentage =  14.29
Row  21148  no. of null values = 2 percentage =  9.52
Row  21149  no. of null values = 1 percentage =  4.76
Row  21150  no. of null values = 3 percentage =  14.29
Row  21151  no. of null values = 5 percentage =  23.81
Row  21152  no. of null values = 1 percentage =  4.76
Row  21153  no. of null values = 1 percentage =  4.76
Row  21154  no. of null values = 3 percentage =  14.29
Row  21155  no. of null values = 2 percentage =  9.52
Row  21156  no. of null values = 3 percentage =  14.29
Row  21157  no. of null values = 2 percentage =  9.52
Row  21158  no. of null values = 5 percentage =  23.81
Row  21159  no. of null values = 3 percentage =  14.29
Row  21160  no. of null values = 3 percentage =  14.29
Row  21161  no. of null values = 3 percentage =  14.29
Row  21162  no. of null values = 2 percentage =  9.52
Row  21163  no. of

Row  21322  no. of null values = 4 percentage =  19.05
Row  21323  no. of null values = 5 percentage =  23.81
Row  21324  no. of null values = 2 percentage =  9.52
Row  21325  no. of null values = 3 percentage =  14.29
Row  21326  no. of null values = 4 percentage =  19.05
Row  21327  no. of null values = 2 percentage =  9.52
Row  21328  no. of null values = 4 percentage =  19.05
Row  21329  no. of null values = 2 percentage =  9.52
Row  21330  no. of null values = 3 percentage =  14.29
Row  21331  no. of null values = 4 percentage =  19.05
Row  21332  no. of null values = 3 percentage =  14.29
Row  21333  no. of null values = 3 percentage =  14.29
Row  21334  no. of null values = 4 percentage =  19.05
Row  21335  no. of null values = 1 percentage =  4.76
Row  21336  no. of null values = 2 percentage =  9.52
Row  21337  no. of null values = 3 percentage =  14.29
Row  21338  no. of null values = 2 percentage =  9.52
Row  21339  no. of null values = 1 percentage =  4.76
Row  21340  no. o

Row  21484  no. of null values = 3 percentage =  14.29
Row  21485  no. of null values = 4 percentage =  19.05
Row  21486  no. of null values = 2 percentage =  9.52
Row  21487  no. of null values = 9 percentage =  42.86
Row  21488  no. of null values = 3 percentage =  14.29
Row  21489  no. of null values = 3 percentage =  14.29
Row  21490  no. of null values = 2 percentage =  9.52
Row  21491  no. of null values = 4 percentage =  19.05
Row  21492  no. of null values = 1 percentage =  4.76
Row  21493  no. of null values = 3 percentage =  14.29
Row  21494  no. of null values = 3 percentage =  14.29
Row  21495  no. of null values = 3 percentage =  14.29
Row  21496  no. of null values = 2 percentage =  9.52
Row  21497  no. of null values = 9 percentage =  42.86
Row  21498  no. of null values = 5 percentage =  23.81
Row  21499  no. of null values = 4 percentage =  19.05
Row  21500  no. of null values = 4 percentage =  19.05
Row  21501  no. of null values = 4 percentage =  19.05
Row  21502  no

Row  21670  no. of null values = 9 percentage =  42.86
Row  21671  no. of null values = 4 percentage =  19.05
Row  21672  no. of null values = 1 percentage =  4.76
Row  21673  no. of null values = 3 percentage =  14.29
Row  21674  no. of null values = 9 percentage =  42.86
Row  21675  no. of null values = 4 percentage =  19.05
Row  21676  no. of null values = 1 percentage =  4.76
Row  21677  no. of null values = 1 percentage =  4.76
Row  21678  no. of null values = 3 percentage =  14.29
Row  21679  no. of null values = 9 percentage =  42.86
Row  21680  no. of null values = 1 percentage =  4.76
Row  21681  no. of null values = 3 percentage =  14.29
Row  21682  no. of null values = 3 percentage =  14.29
Row  21683  no. of null values = 2 percentage =  9.52
Row  21684  no. of null values = 3 percentage =  14.29
Row  21685  no. of null values = 2 percentage =  9.52
Row  21686  no. of null values = 3 percentage =  14.29
Row  21687  no. of null values = 1 percentage =  4.76
Row  21688  no. o

Row  21873  no. of null values = 3 percentage =  14.29
Row  21874  no. of null values = 3 percentage =  14.29
Row  21875  no. of null values = 1 percentage =  4.76
Row  21876  no. of null values = 2 percentage =  9.52
Row  21877  no. of null values = 4 percentage =  19.05
Row  21878  no. of null values = 2 percentage =  9.52
Row  21879  no. of null values = 3 percentage =  14.29
Row  21880  no. of null values = 1 percentage =  4.76
Row  21881  no. of null values = 3 percentage =  14.29
Row  21882  no. of null values = 4 percentage =  19.05
Row  21883  no. of null values = 2 percentage =  9.52
Row  21884  no. of null values = 9 percentage =  42.86
Row  21885  no. of null values = 3 percentage =  14.29
Row  21886  no. of null values = 2 percentage =  9.52
Row  21887  no. of null values = 4 percentage =  19.05
Row  21888  no. of null values = 3 percentage =  14.29
Row  21889  no. of null values = 2 percentage =  9.52
Row  21890  no. of null values = 3 percentage =  14.29
Row  21891  no. o

Row  22057  no. of null values = 2 percentage =  9.52
Row  22058  no. of null values = 5 percentage =  23.81
Row  22059  no. of null values = 4 percentage =  19.05
Row  22060  no. of null values = 7 percentage =  33.33
Row  22061  no. of null values = 2 percentage =  9.52
Row  22062  no. of null values = 3 percentage =  14.29
Row  22063  no. of null values = 9 percentage =  42.86
Row  22064  no. of null values = 1 percentage =  4.76
Row  22065  no. of null values = 2 percentage =  9.52
Row  22066  no. of null values = 2 percentage =  9.52
Row  22067  no. of null values = 4 percentage =  19.05
Row  22068  no. of null values = 3 percentage =  14.29
Row  22069  no. of null values = 1 percentage =  4.76
Row  22070  no. of null values = 1 percentage =  4.76
Row  22071  no. of null values = 1 percentage =  4.76
Row  22072  no. of null values = 1 percentage =  4.76
Row  22073  no. of null values = 3 percentage =  14.29
Row  22074  no. of null values = 9 percentage =  42.86
Row  22075  no. of 

Row  22300  no. of null values = 4 percentage =  19.05
Row  22301  no. of null values = 9 percentage =  42.86
Row  22302  no. of null values = 4 percentage =  19.05
Row  22303  no. of null values = 4 percentage =  19.05
Row  22304  no. of null values = 1 percentage =  4.76
Row  22305  no. of null values = 3 percentage =  14.29
Row  22306  no. of null values = 3 percentage =  14.29
Row  22307  no. of null values = 3 percentage =  14.29
Row  22308  no. of null values = 4 percentage =  19.05
Row  22309  no. of null values = 2 percentage =  9.52
Row  22310  no. of null values = 1 percentage =  4.76
Row  22311  no. of null values = 2 percentage =  9.52
Row  22312  no. of null values = 3 percentage =  14.29
Row  22313  no. of null values = 5 percentage =  23.81
Row  22314  no. of null values = 4 percentage =  19.05
Row  22315  no. of null values = 3 percentage =  14.29
Row  22316  no. of null values = 4 percentage =  19.05
Row  22317  no. of null values = 2 percentage =  9.52
Row  22318  no.

Row  22608  no. of null values = 3 percentage =  14.29
Row  22609  no. of null values = 1 percentage =  4.76
Row  22610  no. of null values = 9 percentage =  42.86
Row  22611  no. of null values = 2 percentage =  9.52
Row  22612  no. of null values = 4 percentage =  19.05
Row  22613  no. of null values = 4 percentage =  19.05
Row  22614  no. of null values = 4 percentage =  19.05
Row  22615  no. of null values = 1 percentage =  4.76
Row  22616  no. of null values = 4 percentage =  19.05
Row  22617  no. of null values = 4 percentage =  19.05
Row  22618  no. of null values = 4 percentage =  19.05
Row  22619  no. of null values = 3 percentage =  14.29
Row  22620  no. of null values = 4 percentage =  19.05
Row  22621  no. of null values = 9 percentage =  42.86
Row  22622  no. of null values = 1 percentage =  4.76
Row  22623  no. of null values = 2 percentage =  9.52
Row  22624  no. of null values = 4 percentage =  19.05
Row  22625  no. of null values = 2 percentage =  9.52
Row  22626  no. 

Row  22791  no. of null values = 1 percentage =  4.76
Row  22792  no. of null values = 3 percentage =  14.29
Row  22793  no. of null values = 4 percentage =  19.05
Row  22794  no. of null values = 2 percentage =  9.52
Row  22795  no. of null values = 2 percentage =  9.52
Row  22796  no. of null values = 3 percentage =  14.29
Row  22797  no. of null values = 3 percentage =  14.29
Row  22798  no. of null values = 2 percentage =  9.52
Row  22799  no. of null values = 3 percentage =  14.29
Row  22800  no. of null values = 1 percentage =  4.76
Row  22801  no. of null values = 9 percentage =  42.86
Row  22802  no. of null values = 3 percentage =  14.29
Row  22803  no. of null values = 1 percentage =  4.76
Row  22804  no. of null values = 1 percentage =  4.76
Row  22805  no. of null values = 3 percentage =  14.29
Row  22806  no. of null values = 3 percentage =  14.29
Row  22807  no. of null values = 9 percentage =  42.86
Row  22808  no. of null values = 2 percentage =  9.52
Row  22809  no. of

Row  22950  no. of null values = 2 percentage =  9.52
Row  22951  no. of null values = 2 percentage =  9.52
Row  22952  no. of null values = 3 percentage =  14.29
Row  22953  no. of null values = 4 percentage =  19.05
Row  22954  no. of null values = 5 percentage =  23.81
Row  22955  no. of null values = 9 percentage =  42.86
Row  22956  no. of null values = 2 percentage =  9.52
Row  22957  no. of null values = 2 percentage =  9.52
Row  22958  no. of null values = 3 percentage =  14.29
Row  22959  no. of null values = 4 percentage =  19.05
Row  22960  no. of null values = 1 percentage =  4.76
Row  22961  no. of null values = 3 percentage =  14.29
Row  22962  no. of null values = 4 percentage =  19.05
Row  22963  no. of null values = 4 percentage =  19.05
Row  22964  no. of null values = 1 percentage =  4.76
Row  22965  no. of null values = 2 percentage =  9.52
Row  22966  no. of null values = 9 percentage =  42.86
Row  22967  no. of null values = 4 percentage =  19.05
Row  22968  no. o

Row  23157  no. of null values = 3 percentage =  14.29
Row  23158  no. of null values = 1 percentage =  4.76
Row  23159  no. of null values = 2 percentage =  9.52
Row  23160  no. of null values = 2 percentage =  9.52
Row  23161  no. of null values = 3 percentage =  14.29
Row  23162  no. of null values = 3 percentage =  14.29
Row  23163  no. of null values = 1 percentage =  4.76
Row  23164  no. of null values = 2 percentage =  9.52
Row  23165  no. of null values = 3 percentage =  14.29
Row  23166  no. of null values = 9 percentage =  42.86
Row  23167  no. of null values = 4 percentage =  19.05
Row  23168  no. of null values = 2 percentage =  9.52
Row  23169  no. of null values = 2 percentage =  9.52
Row  23170  no. of null values = 2 percentage =  9.52
Row  23171  no. of null values = 3 percentage =  14.29
Row  23172  no. of null values = 2 percentage =  9.52
Row  23173  no. of null values = 4 percentage =  19.05
Row  23174  no. of null values = 2 percentage =  9.52
Row  23175  no. of n

Row  23451  no. of null values = 1 percentage =  4.76
Row  23452  no. of null values = 1 percentage =  4.76
Row  23453  no. of null values = 2 percentage =  9.52
Row  23454  no. of null values = 2 percentage =  9.52
Row  23455  no. of null values = 3 percentage =  14.29
Row  23456  no. of null values = 1 percentage =  4.76
Row  23457  no. of null values = 2 percentage =  9.52
Row  23458  no. of null values = 4 percentage =  19.05
Row  23459  no. of null values = 1 percentage =  4.76
Row  23460  no. of null values = 2 percentage =  9.52
Row  23461  no. of null values = 2 percentage =  9.52
Row  23462  no. of null values = 2 percentage =  9.52
Row  23463  no. of null values = 1 percentage =  4.76
Row  23464  no. of null values = 10 percentage =  47.62
Row  23465  no. of null values = 4 percentage =  19.05
Row  23466  no. of null values = 4 percentage =  19.05
Row  23467  no. of null values = 1 percentage =  4.76
Row  23468  no. of null values = 9 percentage =  42.86
Row  23469  no. of nu

#### 9. Determine the column names having more than 30% missing values and remove them from the dataset.

In [63]:
for i in data.columns:
    null=round(((data[i].isnull().sum()/data.shape[0])*100),2)
    if null >30:
        print('% of null values in',i,'= ',null)
        #Drop columns having null null value
        data.drop([i],axis=1,inplace=True)

% of null values in BuildingArea =  57.69
% of null values in YearBuilt =  51.26
% of null values in CouncilArea =  33.88


#### 10. Determine the number and percentage of rows having more than 5 missing values. If the percentage of such rows is less than 20% then remove all these rows.

In [64]:
count=data.apply(lambda x: abs(x.count()-data.shape[1]), axis=1)    
row_id=count.index.tolist()
#null value
values=count.values.tolist() 

for i in range(len(row_id)):
    if values[i]>5:
        null_values=values[i]  #number of null values
        percentage=round(((null_values/data.shape[1])*100),2)
        if percentage<20:
            print('At row number ',row_id[i],' number of null values =',null_values,'percentage= ',percentage)
            print('Removing row number',row_id[i])
            df10.dropna(axis=0,inplace=True)   
else:
    print('No Rows exist with both condition satisfied')
    data.shape

No Rows exist with both condition satisfied


#### 11. Display the description of data and percentage of missing values still left column wise.

In [65]:
data.describe(include='object')

Suburb       Address   Type Method SellerG        Date  \
count       23547         23547  23547  23547   23547       23547   
unique        336         23108      3      9     330          58   
top     Reservoir  5 Charles St      h      S  Nelson  27-05-2017   
freq          629             5  15760  13660    2374         770   

                   Regionname  
count                   23546  
unique                      8  
top     Southern Metropolitan  
freq                     8772

In [66]:
for i in data.columns:
    null=round(((data[i].isnull().sum()/data.shape[0])*100),2)
    if null >0:
        print('percentage of null values in',i,'= ',null)

percentage of null values in Suburb =  0.55
percentage of null values in Address =  0.55
percentage of null values in Rooms =  0.55
percentage of null values in Type =  0.55
percentage of null values in Price =  22.31
percentage of null values in Method =  0.55
percentage of null values in SellerG =  0.55
percentage of null values in Date =  0.55
percentage of null values in Distance =  0.56
percentage of null values in Postcode =  0.56
percentage of null values in Bedroom2 =  19.48
percentage of null values in Bathroom =  19.49
percentage of null values in Car =  20.09
percentage of null values in Landsize =  26.47
percentage of null values in Lattitude =  18.73
percentage of null values in Longtitude =  18.73
percentage of null values in Regionname =  0.56
percentage of null values in Propertycount =  0.56


#### 12. Considering the 'Price' column as target variable which is to be predicted, how will you treat the missing values? Give a brief explanation and treat the missing values in 'Price' column.

In [67]:
data.head(7)

Suburb             Address  Rooms Type      Price Method SellerG  \
0  Abbotsford       68 Studley St    2.0    h        NaN     SS  Jellis   
1  Abbotsford        85 Turner St    2.0    h  1480000.0      S  Biggin   
2  Abbotsford     25 Bloomburg St    2.0    h  1035000.0      S  Biggin   
3  Abbotsford  18/659 Victoria St    3.0    u        NaN     VB  Rounds   
4  Abbotsford        5 Charles St    3.0    h  1465000.0     SP  Biggin   
5  Abbotsford    40 Federation La    3.0    h   850000.0     PI  Biggin   
6  Abbotsford         55a Park St    4.0    h  1600000.0     VB  Nelson   

         Date  Distance  Postcode  Bedroom2  Bathroom  Car  Landsize  \
0  03-09-2016       2.5    3067.0       2.0       1.0  1.0     126.0   
1  03-12-2016       2.5    3067.0       2.0       1.0  1.0     202.0   
2  04-02-2016       2.5    3067.0       2.0       1.0  0.0     156.0   
3  04-02-2016       2.5    3067.0       3.0       2.0  1.0       0.0   
4  04-03-2017       2.5    3067.0       3.0       2.0  0.0     134.0   
5  04-03-2017       2.5    3067.0       3.0       2.0  1.0      94.0   
6  04-06-2016       2.5    3067.0       3.0       1.0  2.0     120.0   

   Lattitude  Longtitude             Regionname  Propertycount  
0   -37.8014    144.9958  Northern Metropolitan         4019.0  
1   -37.7996    144.9984  Northern Metropolitan         4019.0  
2   -37.8079    144.9934  Northern Metropolitan         4019.0  
3   -37.8114    145.0116  Northern Metropolitan         4019.0  
4   -37.8093    144.9944  Northern Metropolitan         4019.0  
5   -37.7969    144.9969  Northern Metropolitan         4019.0  
6   -37.8072    144.9941  Northern Metropolitan         4019.0

The values in the price column are continious values so price the null values of the price column can be filler by the mean value of the column.

In [68]:
mean=round(data['Price'].mean())
mean

1056697.0

In [69]:
data['Price'].fillna(mean,inplace=True)

#### 13. Now check the percentage of missing values in each column again, if missing values exist, treat it with the most suitable method.

In [70]:
for i in data.columns:
    null=round(((data[i].isnull().sum()/data.shape[0])*100),2)
    if null >0:
        print('percentage of null values in',i,'= ',null)

percentage of null values in Suburb =  0.55
percentage of null values in Address =  0.55
percentage of null values in Rooms =  0.55
percentage of null values in Type =  0.55
percentage of null values in Method =  0.55
percentage of null values in SellerG =  0.55
percentage of null values in Date =  0.55
percentage of null values in Distance =  0.56
percentage of null values in Postcode =  0.56
percentage of null values in Bedroom2 =  19.48
percentage of null values in Bathroom =  19.49
percentage of null values in Car =  20.09
percentage of null values in Landsize =  26.47
percentage of null values in Lattitude =  18.73
percentage of null values in Longtitude =  18.73
percentage of null values in Regionname =  0.56
percentage of null values in Propertycount =  0.56


In [71]:
for i in data.columns:
    null=round(((data[i].isnull().sum()/data.shape[0])*100),2)
    if null >0:
        if data[i].dtype == 'float64':
            data[i].fillna(data[i].mean(),inplace=True)
            
        if data[i].dtype == 'object':
            data[i].fillna(data[i].mode()[0],inplace=True)

In [72]:
data.isnull().sum()

Suburb           0
Address          0
Rooms            0
Type             0
Price            0
Method           0
SellerG          0
Date             0
Distance         0
Postcode         0
Bedroom2         0
Bathroom         0
Car              0
Landsize         0
Lattitude        0
Longtitude       0
Regionname       0
Propertycount    0
dtype: int64

#### 14. Now since you have completed your data cleaning, determine the number and percentage of columns removed.

In [73]:
print('Earlier the shape of dataset was',prev.shape,'\nAfter datacleaning procedure the shape becomes',data.shape)

Earlier the shape of dataset was (23678, 21) 
After datacleaning procedure the shape becomes (23678, 18)


In [75]:
print('Number of column removed= ',prev.shape[1]-data.shape[1])

Number of column removed=  3


In [76]:
p=round(((1-(data.shape[1]/prev.shape[1]))*100),1)
print('Percentage of column removed= ',p)

Percentage of column removed=  14.3


#### 15. Determine the number and percentage of rows removed in whole process. Also comment briefly that according to you the amount of data lost is justified or not.

In [78]:
print('Number of rows removed= ',prev.shape[0]-data.shape[0])

Number of rows removed=  0


No rows were removed in the process. We were instructed to dropped columns whose null values were higher than 30% and the process column named percentage of null values in BuildingArea,YearBuilt and CouncilArea with null values of 57.69, 51.26, 33.88 percent respectively. So dropping them was a justified move because filling such high values them with their mean or mode put an impact on the variance of the dataset